cli/azd/pkg/commands/pipeline/pipeline_manager.go

```diff
@@ -420,3 +447,45 @@ func (manager *PipelineManager) Configure(ctx context.Context) (
 		PipelineLink:   strings.TrimSuffix(ciPipeline.remote, ".git"),
 	}, nil
 }
+
+func checkRoleAssignments(
+	ctx context.Context,
+	azcli azcli.AzCli,
+	subscriptionId string,
+	principalId string,
+	console input.Console,
+) error {
+	// Find the specified role in the subscription scope
+	scope := azure.SubscriptionRID(subscriptionId)
+	roles, err := azcli.GetUserRoleDefinitionName(ctx, subscriptionId, scope, principalId)
+	if err != nil {
+		return err
+	}
+	for _, name := range roles {
+		if name == "Owner" || name == "User Access Administrator" {
+			return nil
+		}
+	}
+
+	message := output.WithWarningFormat(
+		"WARNING: Your user account does not appear to have either the Owner or User Access Administrator role assigned to it. " +
+			"This operation will attempt to grant permissions to the newly created service principal and this will fail " +
+			"unless you have been granted these permissions with a custom role assignment. " +
+			"Do you want to continue with configuring your pipeline?")
+
+	confirm, err := console.Confirm(ctx, input.ConsoleOptions{
+		Message: message,
+	})
+	if err != nil {
+		return err
+	}
+
+	if !confirm {
+		return fmt.Errorf("missing required user role assignment for authorization: Owner or User Access Administrator. " +
+			"Please contact your Azure subscription administrator to assign required role to your account or" +
+			"login to azd with an account that has required role assignment")
```

Comment:
```suggestion
			" login to azd with an account that has required role assignment")
```

missing a space between 'or' and 'login'